pkg/ddc/jindofsx/transform.go

```diff
@@ -335,53 +340,46 @@ func (e *JindoFSxEngine) transformMaster(runtime *datav1alpha1.JindoRuntime, met
 
 		// support HDFS HA
 		if strings.HasPrefix(mount.MountPoint, "hdfs://") {
+			mountType = "hdfs"
 			for key, value := range mount.Options {
 				propertiesFileStore[strings.Replace(key, "fs", "jindofsx", 1)] = value
 			}
 		}
 
+		fuseOnly := runtime.Spec.Master.Disabled && runtime.Spec.Worker.Disabled
 		// to check whether encryptOptions exist
 		for _, encryptOption := range mount.EncryptOptions {
 			key := encryptOption.Name
 			secretKeyRef := encryptOption.ValueFrom.SecretKeyRef
-			secret, err := kubeclient.GetSecret(e.Client, secretKeyRef.Name, e.namespace)
-			if err != nil {
-				e.Log.Info("can't get the secret")
-				break
-			}
-			value := secret.Data[secretKeyRef.Key]
-			if err != nil {
-				e.Log.Info("decode value failed")
-			}
-			if key == "fs.oss.accessKeyId" {
-				propertiesFileStore["jindofsx.oss.accessKeyId"] = string(value)
-			}
-			if key == "fs.oss.accessKeySecret" {
-				propertiesFileStore["jindofsx.oss.accessKeySecret"] = string(value)
-			}
-			if key == "fs.s3.accessKeyId" {
-				propertiesFileStore["jindofsx.s3.accessKeyId"] = string(value)
-			}
-			if key == "fs.s3.accessKeySecret" {
-				propertiesFileStore["jindofsx.s3.accessKeySecret"] = string(value)
-			}
-			if key == "fs.cos.accessKeyId" {
-				propertiesFileStore["jindofsx.cos.accessKeyId"] = string(value)
-			}
-			if key == "fs.cos.accessKeySecret" {
-				propertiesFileStore["jindofsx.cos.accessKeySecret"] = string(value)
-			}
-			if key == "fs.obs.accessKeyId" {
-				propertiesFileStore["jindofsx.obs.accessKeyId"] = string(value)
-			}
-			if key == "fs.obs.accessKeySecret" {
-				propertiesFileStore["jindofsx.obs.accessKeySecret"] = string(value)
+			if (fuseOnly && strings.Compare(fuseTag, IMAGE_TAG_SUPPORT_AK_FILE) >= 0) || (strings.Compare(smartdataTag, IMAGE_TAG_SUPPORT_AK_FILE) >= 0 && strings.Compare(fuseTag, IMAGE_TAG_SUPPORT_AK_FILE) >= 0) {
```

Comment:
Can we leverage  versiontuils here ?https://github.com/fluid-cloudnative/fluid/blob/master/pkg/ddc/alluxio/transform_fuse.go#L125